packages/evm/src/opcodes/functions.ts

```diff
@@ -984,6 +1011,346 @@ export const handlers: Map<number, OpHandler> = new Map([
+      const typeSection = runState.env.eof!.container.body.typeSections[sectionTarget]
+      if (1024 < stackItems + typeSection?.inputs - typeSection?.maxStackHeight) {
+        trap(EOFError.StackOverflow)
+      }
+      /*if (runState.env.eof!.eofRunState.returnStack.length >= 1024) {
+        trap(EOFError.ReturnStackOverflow)
+      }
+      runState.env.eof?.eofRunState.returnStack.push(runState.programCounter + 2)*/
+
+      // Find out the opcode we should jump into
+      runState.programCounter = runState.env.eof!.container.header.getCodePosition(sectionTarget)
+    },
+  ],
+  // 0xe6: DUPN
+  [
+    0xe6,
+    function (runState, _common) {
+      if (runState.env.eof === undefined) {
+        // Opcode not available in legacy contracts
+        trap(ERROR.INVALID_OPCODE)
+      }
+      const toDup =
+        Number(
+          bytesToBigInt(
+            runState.code.subarray(runState.programCounter, runState.programCounter + 1)
+          )
+        ) + 1
+      runState.stack.dup(toDup)
+      runState.programCounter++
+    },
+  ],
+  // 0xe7: SWAPN
+  [
+    0xe7,
+    function (runState, _common) {
+      if (runState.env.eof === undefined) {
+        // Opcode not available in legacy contracts
+        trap(ERROR.INVALID_OPCODE)
+      }
+      const toSwap =
+        Number(
+          bytesToBigInt(
+            runState.code.subarray(runState.programCounter, runState.programCounter + 1)
+          )
+        ) + 1
+      runState.stack.swap(toSwap)
+      runState.programCounter++
+    },
+  ],
+  // 0xe8: EXCHANGE
+  [
+    0xe8,
+    function (runState, _common) {
+      if (runState.env.eof === undefined) {
+        // Opcode not available in legacy contracts
+        trap(ERROR.INVALID_OPCODE)
+      }
+      const toExchange = Number(
+        bytesToBigInt(runState.code.subarray(runState.programCounter, runState.programCounter + 1))
+      )
+      const n = (toExchange >> 4) + 1
+      const m = (toExchange & 0x0f) + 1
+      runState.stack.exchange(n, n + m)
+      runState.programCounter++
+    },
+  ],
+  // 0xec: EOFCREATE
+  [
+    0xec,
+    async function (runState, _common) {
+      if (runState.env.eof === undefined) {
+        // Opcode not available in legacy contracts
+        trap(ERROR.INVALID_OPCODE)
+      } else {
+        // Read container index
+        const containerIndex = runState.env.code[runState.programCounter]
+        const containerCode = runState.env.eof!.container.body.containerSections[containerIndex]
+
+        // Pop stack values
+        const [value, salt, inputOffset, inputSize] = runState.stack.popN(4)
+
+        const gasLimit = runState.messageGasLimit!
+        runState.messageGasLimit = undefined
+
+        let data = new Uint8Array(0)
+        if (inputSize !== BIGINT_0) {
+          data = runState.memory.read(Number(inputOffset), Number(inputSize), true)
+        }
+
+        runState.programCounter++ // Jump over the immediate byte
```

Comment:
Is this byte the `EOFCREATE` opcode?

Reply:
No, it jumps over the immediate byte, which is the container the EOFCREATE points to. You can compare it with a PUSH1 operation, in that case it would "jump over" the byte which is being pushed to stack.